sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1852,6 +1852,15 @@ object SQLConf {
       .createWithDefault(
         "org.apache.spark.sql.execution.streaming.state.HDFSBackedStateStoreProvider")
 
+  val STATE_STORE_MAINTENANCE_THREADS =
+    buildConf("spark.sql.streaming.stateStore.stateStoreMaintenanceThreads")
+      .internal()
+      .doc("Number of threads in the thread pool that perform clean up and snapshotting tasks " +
+        "for stateful streaming queries.")
+      .intConf
+      .checkValue(_ >= 0, "Must not be negative")
```

Comment:
Makes sense. I don't see the case where people wants to disable the maintenance task at all. Actually, running maintenance task is a must, otherwise snapshotting does not trigger which leads the query to be loaded forever for restoring the state from all deltas.